Nos/Service/PushNotificationService.swift

```diff
@@ -159,16 +159,33 @@ import Combine
     
     /// Builds the string needed for the `content` field in the special 
     private func createRegistrationContent(deviceToken: Data, user: CurrentUser) async throws -> String {
-        let publicKeyHex = currentUser.publicKeyHex
+        guard let publicKeyHex = currentUser.publicKeyHex else {
+            throw DecodingError.valueNotFound(
+                CurrentUser.self,
+                DecodingError.Context(
```

Comment:
Can we make a new `PushNotificationError` type instead of using DecodingError (and later EncodingError)?